In [1]:
import pandas as pd

# Load your data
file_path = 'sales_weather_stock.csv'  # Modify with the correct path to your file
data = pd.read_csv(file_path)

# Convert 'date' to datetime type
data['date'] = pd.to_datetime(data['date'])

# Define the date of the earthquake
earthquake_date = pd.Timestamp('2016-04-16')

# Separate the data into before and after the earthquake
data_pre_eq = data[data['date'] < earthquake_date]
data_post_eq = data[data['date'] > earthquake_date]

# Analyze sales, last price, and volume before and after the earthquake
sales_impact_pre = data_pre_eq[['sales', 'Last Price', 'Volume']].describe()
sales_impact_post = data_post_eq[['sales', 'Last Price', 'Volume']].describe()

# Analyze the impact of weather variables on sales
weather_vars = ['temp', 'precip', 'windspeed', 'cloudcover', 'visibility']
weather_impact_pre = data_pre_eq.groupby('date')[weather_vars + ['sales']].mean()
weather_impact_post = data_post_eq.groupby('date')[weather_vars + ['sales']].mean()

# Hypothesis:
# "The 2016 Ecuador earthquake had a significant impact on sales, stock prices, and trading volumes,
# and this impact was modulated by local weather conditions which may show a shift in patterns post-earthquake
# indicating telecoupling effects where distant seismic events affect local economic and environmental conditions."

print("Sales, Last Price, and Volume impact before the earthquake:")
print(sales_impact_pre)
print("\nSales, Last Price, and Volume impact after the earthquake:")
print(sales_impact_post)
print("\nWeather impact on sales before the earthquake:")
print(weather_impact_pre)
print("\nWeather impact on sales after the earthquake:")
print(weather_impact_post)

Sales, Last Price, and Volume impact before the earthquake:
              sales    Last Price        Volume
count  1.463022e+06  1.463022e+06  1.463022e+06
mean   2.812435e+02  2.381705e+00  6.177302e+04
std    8.560818e+02  2.990583e-01  1.446301e+05
min    0.000000e+00  1.724000e+00  9.300000e+01
25%    0.000000e+00  2.262000e+00  1.258900e+04
50%    6.000000e+00  2.504000e+00  2.969900e+04
75%    1.568570e+02  2.631000e+00  6.574600e+04
max    4.627100e+04  3.212000e+00  3.235357e+06

Sales, Last Price, and Volume impact after the earthquake:
               sales     Last Price         Volume
count  554202.000000  554202.000000  554202.000000
mean      419.960800       1.634383   30232.218650
std      1196.581267       0.137110   70971.621368
min         0.000000       1.371000      48.000000
25%         2.000000       1.547000    5051.000000
50%        22.000000       1.590000   13457.000000
75%       244.000000       1.740000   31398.000000
max    124717.000000       1.916000  862

Telecoupling Hypothesis:

"Significant events such as earthquakes can lead to immediate and profound impacts on local economic activities such as sales and stock market operations, which are further influenced by local environmental factors like weather conditions. These impacts are indicative of telecoupling mechanisms where distant seismic activities (even those not immediately local to the main economic centers) influence market behaviors and consumer patterns in significant ways."

In [2]:
import pandas as pd
import statsmodels.api as sm

# Create the earthquake dummy variable
data['post_earthquake'] = (data['date'] > pd.Timestamp('2016-04-16')).astype(int)

# Select your independent variables and dependent variable for the model
# For sales impact analysis
X_sales = data[['post_earthquake', 'temp', 'precip', 'windspeed', 'cloudcover', 'visibility']]  # Independent variables
X_sales = sm.add_constant(X_sales)  # Adds a constant term to the predictor
y_sales = data['sales']  # Dependent variable

# For stock market activity analysis
X_stock = data[['post_earthquake', 'temp', 'precip', 'windspeed', 'cloudcover', 'visibility']]
X_stock = sm.add_constant(X_stock)
y_price = data['Last Price']
y_volume = data['Volume']

# Fit the regression model for sales
model_sales = sm.OLS(y_sales, X_sales).fit()

# Fit the regression model for stock price
model_price = sm.OLS(y_price, X_stock).fit()

# Fit the regression model for trading volume
model_volume = sm.OLS(y_volume, X_stock).fit()

# Print the summary of the regression results
print("Regression Analysis for Sales:")
print(model_sales.summary())
print("\nRegression Analysis for Last Price:")
print(model_price.summary())
print("\nRegression Analysis for Volume:")
print(model_volume.summary())


Regression Analysis for Sales:
                            OLS Regression Results                            
Dep. Variable:                  sales   R-squared:                       0.015
Model:                            OLS   Adj. R-squared:                  0.015
Method:                 Least Squares   F-statistic:                     5081.
Date:                Sat, 20 Apr 2024   Prob (F-statistic):               0.00
Time:                        17:00:50   Log-Likelihood:            -1.6707e+07
No. Observations:             2017224   AIC:                         3.341e+07
Df Residuals:                 2017217   BIC:                         3.341e+07
Df Model:                           6                                         
Covariance Type:            nonrobust                                         
                      coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------------------------------------------------
const      

H2: Impact of International Oil Prices Post-Disaster

Hypothesis: Changes in international oil prices have a more pronounced impact on Ecuador's stock prices following major natural disasters.

In [3]:
data['post_disaster'] = (data['date'] > '2016-04-16').astype(int)  # Assuming 2016 earthquake date
X = data[['post_disaster', 'oilprice']]
X = sm.add_constant(X)
y = data['Last Price']

model = sm.OLS(y, X).fit()
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:             Last Price   R-squared:                       0.666
Model:                            OLS   Adj. R-squared:                  0.666
Method:                 Least Squares   F-statistic:                 2.011e+06
Date:                Sat, 20 Apr 2024   Prob (F-statistic):               0.00
Time:                        17:23:47   Log-Likelihood:                -34166.
No. Observations:             2017224   AIC:                         6.834e+04
Df Residuals:                 2017221   BIC:                         6.838e+04
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                    coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------
const             2.0540      0.001   3323.048

INTERPRETATION:

The regression results related to Hypothesis H2 provide significant insights into the dynamics between international oil prices, the occurrence of a major natural disaster, and their impact on stock prices in Ecuador. This information is particularly relevant to the study of telecoupling, as it illustrates the interconnectedness of global economic factors and local market reactions.

Interpretation of Regression Results:

Constant: The constant term of 2.0540 indicates the baseline last price when both the post_disaster and oilprice variables are at zero. This is a theoretical baseline since oil prices cannot be zero.
Post_Disaster Coefficient (-0.6266): The negative coefficient for the post_disaster dummy variable suggests that following the disaster, the last stock prices were on average 0.6266 units lower than before the disaster, holding oil prices constant. This significant drop (p < 0.000) indicates a pronounced negative impact of the disaster on stock prices. This might be due to increased uncertainty, damage to infrastructure, or disruptions in local economic activities which typically depress investor confidence and market valuations.

Oilprice Coefficient (0.0043): The positive coefficient associated with the oil price suggests that for every unit increase in oil price, the last stock price increases by 0.0043 units, ceteris paribus. This is also highly statistically significant (p < 0.000), reflecting the sensitivity of Ecuador’s economy to fluctuations in oil prices. Given Ecuador's status as an oil-exporting country, rises in oil prices could boost economic prospects, enhancing company earnings reflected in stock prices.
R-squared (0.666): The model explains approximately 66.6% of the variance in the last stock prices, which is quite substantial, indicating that the model fits the data well. The adjustments for the number of predictors (Adj. R-squared) confirm the robustness of the model.

Telecoupling Implications:

The regression results underscore the telecoupling concept, where local economic outcomes in Ecuador are influenced by global economic events (like changes in international oil prices) and significant local natural disasters. The observed decrease in stock prices post-disaster highlights how local events (e.g., earthquakes) can have immediate and profound impacts on economic indicators, which are observable even when accounting for global influences like oil prices.

This dual influence of local and global factors is emblematic of telecoupling mechanisms, where distant socioeconomic and environmental interactions affect local systems. The ability of international oil prices to impact local stock markets signifies the interconnectedness and dependency of local economies on global commodity markets. Such insights can help policymakers and businesses in Ecuador and similar economies better prepare and adapt to both local and global shocks, ensuring more resilient economic planning and response strategies.

These findings also provide a platform for further research into how other global factors (like geopolitical events, global demand shifts) and local factors (like policy changes, economic reforms) interact and shape economic realities in interconnected regions, further exploring the depths of telecoupling in global economic systems.

H3: Weather Patterns and Agricultural Sales

Hypothesis: Changes in weather patterns (temperature and precipitation) significantly impact sales in the agricultural sector in Ecuador.

In [5]:
# Create a specific column for agricultural sales in the 'GROCERY I' category
data['agricultural_sales'] = data.loc[data['family'] == 'GROCERY I', 'sales']

# Prepare your independent variables (exog)
X = data[['temp', 'precip']]  # Make sure these columns don't contain NaN values in rows where 'family' is 'GROCERY I'
X = sm.add_constant(X)  # Adds a constant term to the predictor

# Ensure alignment and remove any rows with NaN values in either X or y
common_index = X.dropna().index.intersection(data['agricultural_sales'].dropna().index)
X_aligned = X.loc[common_index]
y_aligned = data['agricultural_sales'].loc[common_index]

# Fit the regression model
model = sm.OLS(y_aligned, X_aligned).fit()

# Print the model summary
print(model.summary())


                            OLS Regression Results                            
Dep. Variable:     agricultural_sales   R-squared:                       0.097
Model:                            OLS   Adj. R-squared:                  0.097
Method:                 Least Squares   F-statistic:                     3266.
Date:                Sat, 20 Apr 2024   Prob (F-statistic):               0.00
Time:                        17:38:58   Log-Likelihood:            -5.6101e+05
No. Observations:               61128   AIC:                         1.122e+06
Df Residuals:                   61125   BIC:                         1.122e+06
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const       5467.2600     28.097    194.583      0.0

In the regression analysis we discussed, global climate change manifests in the statistical significance and impact of local weather variables (temperature and precipitation) on agricultural sales. Here's a detailed breakdown of how this connection to global climate change is reflected in the data and analysis:

### 1. **Temperature Impact**

- **Negative Correlation**: The negative coefficient for temperature suggests that as temperatures rise, agricultural sales decrease. This pattern could reflect the broader impacts of global warming, which might include:
  - **Heat Stress on Crops**: Higher temperatures can stress plants, reduce yields, and affect crop quality, especially in regions where crops are at the upper limit of their heat tolerance.
  - **Water Scarcity**: Increased temperatures often lead to higher evaporation rates, which can exacerbate water scarcity and negatively affect irrigation-dependent agricultural practices.

### 2. **Precipitation Impact**

- **Positive Correlation**: The positive impact of increased precipitation on agricultural sales suggests that water availability is a crucial factor for crop productivity. In the context of global climate change:
  - **Rainfall Variability**: Changes in global climate patterns can alter precipitation regimes, leading to more intense and less predictable rainfall events. While more rain can benefit crops by alleviating water stress, extreme rainfall can lead to flooding and soil erosion, potentially harming agricultural output.
  - **Adaptation to Changing Rain Patterns**: Areas that adapt to these changes, perhaps through improved water management systems or crop choices suited to variable conditions, may see less negative or even positive impacts from increased rainfall.

### 3. **Reflection of Global Climate Patterns**

- **Local Weather Fluctuations**: Global climate change influences local weather patterns. The significant effects of temperature and precipitation on agriculture as shown in your analysis reflect this influence, indicating how local agricultural markets are inherently linked to broader climatic conditions.
- **Economic Sensitivity to Environmental Changes**: The agricultural sector's sensitivity to climate conditions underscores its vulnerability to global climate change. This sector must continuously adapt to shifting environmental baselines driven by global processes.

### 4. **Telecoupling Considerations**

- **Economic Interdependence**: The study highlights telecoupling through economic interdependence between global climate phenomena and local agricultural outcomes. Changes in global climate directly influence local agricultural productivity, which in turn affects local and potentially global markets.
- **Policy Implications**: Understanding these dynamics is crucial for formulating effective agricultural policies and strategies for climate adaptation. This includes developing resilient agricultural practices that can withstand both the current variability and future climatic conditions anticipated as the global climate continues to change.

### Conclusion

The regression results reflect broader themes of how global climate change impacts local economies, specifically through the lens of agriculture. They underscore the need for robust climate adaptation strategies in agricultural planning and management to mitigate the adverse effects of increased temperatures and leverage beneficial aspects of increased precipitation. This analysis can serve as a foundational piece for deeper investigations into specific adaptive measures and the evaluation of their economic viability and effectiveness in the face of ongoing global climate change.

H4: Geographic Disparity in Disaster Impact

Hypothesis: Urban and rural areas in Ecuador experience different economic impacts from natural disasters, reflected in sales data.

Based on the store information you've provided, which includes cities and states in Ecuador, we can categorize these locations as potentially urban or rural. Generally, larger cities like Quito, Guayaquil, and Cuenca might be considered urban, while smaller towns or those known to be in less densely populated areas could be considered rural. This categorization can help provide insights into the sales patterns post-disaster in urban vs. rural areas.

Without direct knowledge of Ecuador's geography or additional data like population size, economic activity, or density, the urban vs. rural categorization might be somewhat subjective. However, we can generally assume that capital cities and provincial capitals are urban.

Here's how you could update your Python code using this information:

Categorize each city as 'urban' or 'rural' based on your knowledge or assumptions about their size and economic activity.
Merge this categorization with your main dataset.
Run a regression analysis to see if the post-disaster effect on sales differs between these categories.

In [9]:
# Load your sales data
sales_data = pd.read_csv('sales_weather_stock.csv')
sales_data['date'] = pd.to_datetime(sales_data['date'])
sales_data['post_disaster'] = (sales_data['date'] > '2016-04-16').astype(int)

# Here's the categorization based on the cities you've provided
# Note: This is a simplification and might not reflect the actual urban/rural status of these locations.
urban_rural_dict = {
    'Quito': 'urban', 'Guayaquil': 'urban', 'Cuenca': 'urban',  # and so on for the rest of the cities
    # Other cities could be classified as 'rural'
}

# Apply the categorization to the stores data
sales_data['urban_rural'] = sales_data['city'].map(urban_rural_dict).fillna('rural')

# Convert urban/rural to a binary indicator for the regression analysis
sales_data['urban_rural'] = sales_data['urban_rural'].map({'urban': 1, 'rural': 0})

# Prepare your independent variables (exog)
X = sales_data[['urban_rural', 'post_disaster']]
X = sm.add_constant(X)  # Adds a constant term to the predictor

# Prepare your dependent variable (endog)
y = sales_data['sales']

# Fit the regression model
model = sm.OLS(y, X).fit()

# Print the model summary
print(model.summary())


                            OLS Regression Results                            
Dep. Variable:                  sales   R-squared:                       0.012
Model:                            OLS   Adj. R-squared:                  0.012
Method:                 Least Squares   F-statistic:                 1.272e+04
Date:                Sat, 20 Apr 2024   Prob (F-statistic):               0.00
Time:                        18:01:21   Log-Likelihood:            -1.6710e+07
No. Observations:             2017224   AIC:                         3.342e+07
Df Residuals:                 2017221   BIC:                         3.342e+07
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                    coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------
const           186.5325      1.074    173.617

The OLS regression results for Hypothesis H4 provide information on how sales were impacted by the categorization of urban vs. rural areas and the occurrence of a disaster (presumably an earthquake), reflecting elements of telecoupling.

### Interpretation of Regression Results:

- **Constant (186.5325)**: The model suggests that in the absence of the disaster (pre-disaster) and without considering urban or rural classification, the average sales were about 186.53 units.

- **Urban_Rural Coefficient (176.3582)**: This coefficient indicates that urban areas, on average, have 176.36 more sales units compared to rural areas, holding the disaster effect constant. This result is statistically significant (p < 0.000) and could reflect higher population densities, greater economic activity, or a combination of factors that typically characterize urban environments.

- **Post_Disaster Coefficient (138.7173)**: The positive and significant coefficient for the post_disaster variable suggests that after the disaster, sales increased by about 138.72 units, on average, across both urban and rural areas. This could be due to increased demand for goods in response to the disaster, such as emergency supplies or rebuilding materials.

- **Model Fit (R-squared: 0.012)**: The R-squared value indicates that the model explains only 1.2% of the variance in sales, which is relatively low. This suggests that other factors not included in the model also play a significant role in determining sales.

### Implications for Telecoupling:

- **Urban vs. Rural Dynamics**: The significant difference in sales between urban and rural areas could be indicative of how geographic and socioeconomic factors are interconnected across regions within Ecuador. Urban areas may have more resilient economic structures that enable them to maintain higher sales levels, even in the face of disruptive events.

- **Disaster Response**: The increase in sales post-disaster reflects the interconnectedness of environmental shocks and economic responses. The response to a disaster may involve increased economic activity as communities rebuild and recover, demonstrating how local events have ripple effects throughout the economic system.

- **Global and Local Interactions**: The findings hint at telecoupling mechanisms where local disasters elicit responses influenced by broader socioeconomic systems. This might include aid flows, changes in trade patterns, or shifts in resource allocation, all of which are aspects of how distant socioeconomic interactions affect local outcomes.

### Recommendations:

To further refine this analysis, consider the following:

- **Additional Variables**: Introduce more variables into the model that may affect sales, such as regional economic indicators, demographic data, or specific disaster impact metrics.
- **Interaction Term**: Include interaction terms between `post_disaster` and `urban_rural` to see if the effect of the disaster differed between urban and rural settings.
- **Temporal Dynamics**: Look into the sales trends over time, possibly by incorporating time-series analysis to account for temporal patterns in sales data.

In conclusion, the analysis shows that both urbanization and disaster occurrence are significant determinants of sales. However, due to the low R-squared, the model leaves much of the variation in sales unexplained, suggesting the need to consider additional factors or more complex models to fully capture the telecoupling effects on economic activities in Ecuador.

Certainly! Below is a tabular representation of the systems, flows, agents, causes, and effects based on the analyses done for Hypotheses H1 to H4. These elements are key components in the telecoupling framework, which helps in understanding and describing interactions across distance in coupled human and natural systems.

| Component  | H1 Description | H2 Description | H3 Description | H4 Description |
|------------|----------------|----------------|----------------|----------------|
| **Hypothesis** | Increase in sales due to an earthquake, reflecting emergency buying. | Influence of international oil prices on Ecuador's stock market post-disaster. | Impact of weather patterns on agricultural sales in Ecuador. | Differential impacts of a disaster on sales in different Ecuadorian cities/states. |
| **Systems**    | Local markets and communities affected by the earthquake. | Ecuador's stock market and its reaction to global oil prices and local disaster events. | Agricultural sector and its sensitivity to local weather conditions. | Sales systems across different urban and rural areas within Ecuador. |
| **Flows**      | Sales transactions and changes in demand for goods and services. | Financial capital influenced by international oil prices and local economic stability. | Agricultural products moving from farms to markets. | Money, goods, and services flowing in response to a disaster event. |
| **Agents**     | Consumers, retailers, wholesalers, emergency services, and government bodies. | Stock market investors, oil industry players, government regulators, economic policymakers. | Farmers, agribusiness companies, distributors, and consumers. | Retailers, customers, local governments, and disaster relief organizations. |
| **Causes**     | The earthquake serving as a natural disaster trigger. | Global oil price dynamics combined with the effects of a local disaster. | Local climate and weather conditions, potentially linked to broader climate patterns. | The occurrence of a disaster and its varied impact across different geographical areas. |
| **Effects**    | An uptick in sales figures, likely due to increased demand for emergency supplies. | Alterations in stock prices, which may influence investment and economic policy. | Variations in agricultural sales, with potential consequences for food security. | Variance in sales performance, potentially affecting regional economic resilience. |

### Explanatory Notes

- **Systems**: Represent the distinct but interconnected socioeconomic and environmental domains within which the telecoupling occurs. For example, in H1, the system is the local market and community structure that's impacted by the earthquake.

- **Flows**: Refer to the movement or transfer of resources, information, or influence among systems. In H2, this is exemplified by the flow of economic influence from international oil prices to local market conditions.

- **Agents**: Are the individuals, organizations, or entities that operate within or across systems and can influence or be influenced by flows. Agents like consumers and retailers are crucial in the analysis of H1.

- **Causes**: Are the driving forces or initiating events that precipitate changes or interactions between systems. The earthquake itself is the cause in H1, while global oil price changes serve as a cause in H2.

- **Effects**: Are the outcomes or impacts observed within systems as a result of telecoupling interactions. In H3, the effect is the change in agricultural sales due to varying weather patterns.

Each hypothesis explores a distinct aspect of telecoupling, revealing how local and distant events and factors are interlinked, thereby influencing economic activities in particular sectors or regions. These tables could be further elaborated with specific data and findings from the analysis to highlight the detailed interactions and dependencies within the telecoupling framework.